dashboard/app/controllers/foorm/library_questions_controller.rb

```diff
@@ -1,24 +1,24 @@
 module Foorm
-  # Foorm Library Question Editor is only available on levelbuilder or test, for those with levelbuilder permissions.
+  # Foorm Library Editor is only available on levelbuilder or test, for those with levelbuilder permissions.
   class LibraryQuestionsController < ApplicationController
     before_action :require_levelbuilder_mode_or_test_env
     before_action :authenticate_user!
-    authorize_resource
+    load_and_authorize_resource
 
-    # GET '/foorm/library_questions/editor'
-    def editor
-      formatted_names_and_versions = Foorm::LibraryQuestion.all.map {|library_question| {name: library_question.library_name, version: library_question.library_version}}.uniq
-      formatted_names_and_versions.sort_by! {|library| library[:name]}
-      categories = formatted_names_and_versions.map {|data| data[:name].slice(0, data[:name].rindex('/'))}.uniq
+    # GET /foorm/library_questions/:id
+    def show
+      render json: @library_question.to_json
+    end
 
-      @script_data = {
-        props: {
-          libraryNamesAndVersions: formatted_names_and_versions,
-          libraryCategories: categories
-        }.to_json
-      }
+    # PUT /foorm/library_questions/:id/update
+    def update
+      @library_question.question = JSON.pretty_generate(params[:question].as_json)
```

Comment:
What format do you expect the question param to be in?  Do we need to validate it in any way?

Reply:
I am still wrapping my head around the de/serialization here, but in the current editor (which heavily "inspired" this PR) we send a [stringified JS object](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/pd/foorm/editor/FoormSaveBar.jsx#L178), which then gets parsed by Rails into a ActionController::Parameters object. Some [discussion of this question](https://github.com/code-dot-org/code-dot-org/pull/37869#discussion_r524850629) is in the PR where the initial editor controller work was done.

Having already started the front end work (branched off this branch), I've already had to make a couple tweaks to controller methods to work properly. So there may be small updates to these methods as they're put in use.

Re: validation, we do validate before saving a library question using the [method here](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/foorm/library_question.rb#L26). Is there more validation you're envisioning?

Reply:
Thanks for the context!  I was curious because it looks like this code is not doing a lot of input checking (at least not initially) with user-provided input.  Normally, I'd want to see test cases that verify behavior when params[:question] is null, empty, or malformed.  However, given that this API is restricted to levelbuilders, the existing validation is probably enough.